src/object/cli_obj.c

```diff
@@ -2021,15 +2021,22 @@ obj_req_get_tgts(struct dc_object *obj, int *shard, daos_key_t *dkey,
 		D_ASSERT(shard != NULL);
 		D_ASSERT((int)*shard != -1);
 		shard_idx = *shard;
-		if (obj_auxi->is_ec_obj && !to_leader) {
+		if (obj_auxi->is_ec_obj) {
 			struct daos_oclass_attr	*oca;
 
-			/* Client dispatch for EC recx enumeration */
-			flags = OBJ_TGT_FLAG_CLI_DISPATCH;
 			oca = daos_oclass_attr_find(obj->cob_md.omd_id);
 			D_ASSERT(oca != NULL);
-			shard_cnt = obj_ec_data_tgt_nr(oca);
+			if (shard_idx % obj_ec_tgt_nr(oca) <
+			    obj_ec_data_tgt_nr(oca)) {
```

Comment:
to_leader is specified by user to forward the request to the DTX leader, which is used by rebuild or some other cases.
For ec object enumeration, if the user request from the leader, then it choose from the leader, otherwise it will choose from one of parity nodes randomly.

Reply:
ok, thanks. I just was curious why remove to_leader flag checking at L2024?